Sources/Valet/SecureEnclaveValet.swift

```diff
@@ -295,3 +400,97 @@ extension SecureEnclaveValet {
     }
 
 }
+
+// MARK: - Private Extensions
+
+extension SecureEnclaveAccessControl {
+
+    /// The `LAPolicy` that the `SecureEnclaveAccessControl` corresponds to.
+    #if !os(watchOS)
+    @available(tvOS 10.0, *)
+    fileprivate var policy: LAPolicy {
+        switch self {
+        case .userPresence, .devicePasscode:
+            return .deviceOwnerAuthentication
+        case .biometricAny, .biometricCurrentSet:
+            if #available(macOSApplicationExtension 10.12.2, *) {
+                return .deviceOwnerAuthenticationWithBiometrics
+            } else {
+                return .deviceOwnerAuthentication
+            }
+        }
+    }
+    #endif
+
+}
+
+// MARK: - Internal Types
+
+/// A type that statically takes an `LAError` and a `SecureEnclaveAccessControl`
+/// and transforms it to either a `KeychainError` or `SecureEnclaveError`
+enum LAErrorTransformer {
```

Comment:
@dfed @NickEntin here's what I came up with to try and minimize the errors we expose. It feels a little heavy, but I like explicitly handling every `LAError`. But some of those assumptions that we could map directly to `itemNotFound` don't hold if our access didn't require biometry, so those cases I put under `internalError`

Reply:
I like it 👍

Reply:
This seems like a reasonable approach to me